java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java

```diff
@@ -152,24 +186,48 @@ private static boolean hasDiagnosticErrors(CompilationInfo info, Tree tree) {
     //filter anonymous class and intersection types
     private static boolean isValidType(HintContext ctx) {
         TreePath treePath = ctx.getPath();
-        TypeMirror variableTypeMirror = ctx.getInfo().getTrees().getElement(treePath).asType();
-
-        if (Utilities.isAnonymousType(variableTypeMirror)) {
-            return false;
-        } else if (variableTypeMirror.getKind() == TypeKind.DECLARED) {
-            DeclaredType dt = (DeclaredType) variableTypeMirror;
-            if (dt.getTypeArguments().size() > 0) {
-                for (TypeMirror paramType : dt.getTypeArguments()) {
-                    if (Utilities.isAnonymousType(paramType)) {
-                        return false;
+        TreePath initTreePath = ctx.getVariables().get("$init");  //NOI18N
+        TreePath expressionTreePath = ctx.getVariables().get("$expression"); //NOI18N
+        if (initTreePath != null) {
+            TypeMirror variableTypeMirror = ctx.getInfo().getTrees().getElement(treePath).asType();
+            if (Utilities.isAnonymousType(variableTypeMirror)) {
+                return false;
+            } else if (variableTypeMirror.getKind() == TypeKind.DECLARED) {
+                DeclaredType dt = (DeclaredType) variableTypeMirror;
+                if (dt.getTypeArguments().size() > 0) {
+                    for (TypeMirror paramType : dt.getTypeArguments()) {
+                        if (Utilities.isAnonymousType(paramType)) {
+                            return false;
+                        }
                     }
                 }
             }
-        }
 
-        if (!Utilities.isValidType(variableTypeMirror) ||(variableTypeMirror.getKind() == TypeKind.INTERSECTION)) {
+            if (!Utilities.isValidType(variableTypeMirror) ||(variableTypeMirror.getKind() == TypeKind.INTERSECTION)) {
+                return false;
+            }
+            return true;
+        } else if (expressionTreePath != null) {
+            ExecutableElement iterator = findIterable(ctx.getInfo());
+            return !(iterator == null);
+        } else {
             return false;
         }
-        return true;
+    }
+    
+    private static ExecutableElement findIterable(CompilationInfo info) {
```

Comment:
Duplicate code, consider adding to a common utils class.

Reply:
Thanks for the review, Reema. This method was originally present in ExpandEnhancedForLoop.java, I have created a new HintUtils class and moved this method there. I have formatted the new code too.